compiler/rustc_type_ir/src/sty.rs

```diff
@@ -212,6 +216,12 @@ pub enum TyKind<I: Interner> {
 
     /// A placeholder type, used during higher ranked subtyping to instantiate
     /// bound variables.
+    ///
+    /// It is conventional to render anonymous placeholer types like `!N` or `!N_U`,
```

Comment:
this and the comment for bound vars is backwards, its actually `!U_N` and `^D_N` I think?

Reply:
Oh right